assets/src/dashboard/app/views/templateDetails/karma/templateDetails.karma.js

```diff
@@ -0,0 +1,270 @@
+      const viewNextBtn = fixture.screen.getByRole('button', {
+        name: /View next template/,
+      });
+      await fixture.events.click(viewNextBtn);
+
+      // Reparse the current template id from the id query param and assert it's different
+      const { id: nextTemplateId } = getQueryParams();
+
+      const nextTemplate = templates[nextTemplateId];
+
+      // Assert that the rendered title matches the title from state
+      const nextTemplateTitle = utils.getByRole('heading', {
+        name: /Template Title/,
+      });
+      expect(nextTemplateTitle.innerText).toEqual(nextTemplate.title);
+
+      // Click the previous template button and assert we went back to the initial template
+      const viewPreviousBtn = fixture.screen.getByRole('button', {
+        name: /View previous template/,
+      });
+
+      await fixture.events.click(viewPreviousBtn);
+
+      const { id: currentTemplateId } = getQueryParams();
+
+      expect(currentTemplateId).toEqual(initialTemplateId);
+
+      let currentTemplateTitle = utils.getByRole('heading', {
+        name: /Template Title/,
+      });
+
+      expect(currentTemplateTitle.innerText).toEqual(initialTemplate.title);
+    });
+
+    it('should navigate to a related templated when the view button is clicked', async () => {
+      const { templates } = await getTemplatesState();
+      // Parse the current template id from the id query param
+      const { id: initialTemplateId } = getQueryParams();
+      expect(initialTemplateId).toBeTruthy();
+
+      const initialTemplate = templates[initialTemplateId];
+
+      const templateDetailsSection = fixture.screen.getByRole('region', {
+        name: /Template Details/,
+      });
+
+      const templateDetailsUtils = within(templateDetailsSection);
+
+      // Assert that the rendered title matches the title from state
+      const initialTemplateTitle = templateDetailsUtils.getByRole('heading', {
+        name: /Template Title/,
+      });
+      expect(initialTemplateTitle.innerText).toEqual(initialTemplate.title);
+
+      const relatedTemplatesSection = fixture.screen.getByRole('region', {
+        name: /Related Templates/,
+      });
+
+      // Select the first related template (all related templates have the data-testid attribute)
+      const firstRelatedTemplate = relatedTemplatesSection.querySelector(
+        '[data-testid]'
+      );
+
+      // Hover over the first related templated and click the View Button
+      const firstRelatedTemplateUtils = within(firstRelatedTemplate);
+      await fixture.events.hover(firstRelatedTemplate);
+      const view = firstRelatedTemplateUtils.getByText(
+        new RegExp(`^${TEMPLATES_GALLERY_ITEM_CENTER_ACTION_LABELS.template}$`)
+      );
+      await fixture.events.click(view);
+
+      // Assert that the active template has changed
+      const { id: nextTemplateId } = getQueryParams();
+      expect(nextTemplateId).not.toEqual(initialTemplateId);
```

Comment:
This test is failing for me:
```
CUJ: Creator can browse templates in grid view Action: See pre-built template details page should navigate to a related templated when the view button is clicked FAILED
```

`Expected '10' not to equal '10'`

Reply:
This should be passing now, I think GitHub Actions was hiccuping last night or was this failing locally?

Reply:
it was locally failing, fine now though!